### DISCUSSION TITLE
Calling Peaks related question

### DISCUSSION THREAD

**Discussion by sylestiel** [GreenleafLab/ArchR#667] [Questions / Documentation] [answered]
I have a rather naive question pertaining to peak calling. So I understand that we can call peaks using MACS2 or the ArchR supported Tiles Method.

<img width="848" alt="Screen Shot 2021-03-30 at 9 48 57 AM" src="https://user-images.githubusercontent.com/64283689/113008984-2aadd500-913d-11eb-9c34-ba7fc66f0e51.png">

I'm using both scripts as shown above in the same sequence.

I'm perplexed because when I query the project  for the available Matrices it shows the presence of the Tile Matix but the dim on the SummarizedExperiment matches the number of Union Peaks called with MACS2.

Can you help clarify this for me?

**Comment by rcorces** [comment]
> the dim on the SummarizedExperiment matches the number of Union Peaks called with MACS2

What summarized experiment? Can you show the full code you are using?

**Comment by sylestiel** [comment]
<img width="405" alt="Screen Shot 2021-03-31 at 11 06 21 AM" src="https://user-images.githubusercontent.com/64283689/113175546-28b54600-9211-11eb-897b-baff3e0c35e8.png">

<img width="540" alt="image" src="https://user-images.githubusercontent.com/64283689/113175581-323eae00-9211-11eb-88ea-45c7d0e491cd.png">


The MarkersPeak is the Summarized Experiment I was referring to. Sorry for not making that clear in the past.

Anyway its dim: ### 202685 9 
shows that the number of features is the same as from calling peaks using Mac2(Creating Union Peak Set ### 202685). When I used the peakMethod as shown above in the previous post then the Union Peak Set count would have been ### 246637. 


### Making Pseudo-bulk Replicates
```{r}
proj4 <- addGroupCoverages(ArchRProj = proj3, groupBy = "Clusters")
```




### Calling Peaks w/ Macs2
```{r}
proj4 <- addReproduciblePeakSet(
    ArchRProj = proj4, 
    groupBy = "Clusters", 
    pathToMacs2 = "/Applications/NGS_Analysis_Applications/peak_callers/MACS-master/bin/macs2"
)
```

```{r}
getPeakSet(proj4)
```




### Calling Peaks w/ TileMatrix
```{r}
proj4Tmp <- addReproduciblePeakSet(
    ArchRProj = proj4, 
    groupBy = "Clusters",
    peakMethod = "Tiles",
    method = "p", force = TRUE
)
```
### addPeakMatrix
```{r}
proj5 <- addPeakMatrix(proj4)
```

**Reply by rcorces** [reply]
I was hoping you would show how you are creating your MarkerPeaks summarize experiment. I have a feeling you are mixing up `proj4` and `proj4Tmp` when creating this or adding the peak matrix.

**Comment by sylestiel** [comment]
### Calling Peaks w/ Macs2
```{r}
proj4 <- addReproduciblePeakSet(
    ArchRProj = proj4, 
    groupBy = "Clusters", 
    pathToMacs2 = "/Applications/NGS_Analysis_Applications/peak_callers/MACS-master/bin/macs2"
)
```

```{r}
getPeakSet(proj4)
```




### Calling Peaks w/ TileMatrix
```{r}
proj4Tmp <- addReproduciblePeakSet(
    ArchRProj = proj4, 
    groupBy = "Clusters",
    peakMethod = "Tiles",
    method = "p", force = TRUE
)
```

```{r}
getPeakSet(proj4Tmp)
```

proj5 <- addPeakMatrix(proj4)

markersPeaks <- getMarkerFeatures(
    ArchRProj = proj5, 
    useMatrix = "PeakMatrix", 
    groupBy = "Clusters",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  testMethod = "wilcoxon"
)
```
I have the script shown. I just used it as shown in the manual. Can you suggest the appropriate changes?

Thanks!

**Reply by rcorces** [reply]
I'm very confused. 

This:
```
proj4Tmp <- addReproduciblePeakSet(
    ArchRProj = proj4, 
    groupBy = "Clusters",
    peakMethod = "Tiles",
    method = "p", force = TRUE
)
```

creates a *NEW* ArchRProject object called proj4Tmp and does *not* modify `proj4`. So `proj5` of course only has the peakMatrix present in `proj4`

**Comment by sylestiel** [comment]
Yes. I share the confusion. In other words, as is shown in the manual although we create a NEW ArchR project based on Tiles we are essentially not using it in the downstream analysis steps since we are using proj4 and not proj4Tmp.

Therefore, the output from Macs2 is what is being used. So does Macs2 peak calling adhere to the 500bp Tiles method when calling peaks?  Gosh! I sure hope I'm explaining clearly.

**Reply by rcorces** [reply]
> I sure hope I'm explaining clearly.

You aren't :)

**Comment by sylestiel** [comment]
So is it okay to proceed with Macs2 peak calling for downstream steps? Would you still consider it a Tile Matrix?
When should we use proj4Tmp as opposed to proj4

**Comment by rcorces** [comment, marked as answer]
I think you are missing some fundamental understanding from the manual. There are two peak calling methods, one that uses MACS2 which is always recommended. The other ("tiles") is a custom peak calling method that Jeff wrote so that people who were unable to install MACS2 could still proceed with analysis. This method has not been well validated and should not be used unless absolutely necessary. The manual shows how one would theoretically call peaks without MACS2 but stores this in a temporary project because we do not use it for any analyses downstream.

**Comment by sylestiel** [comment]
Thank you!
